### PR TITLE
add com.emqx.MQTTX

### PR DESCRIPTION
**Pull request by ysfscream**
### Please confirm your submission meets all the criteria

- [ ] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.emqx.MQTTX.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+modules:
+  - name: MQTTX
+    buildsystem: simple
+    build-commands:
+      - chmod +x MQTTX.AppImage
+      - ./MQTTX.AppImage --appimage-extract
+      - cp -r squashfs-root/ /app/mqttx
+      - install -D mqttx.sh /app/bin/mqttx.sh
+      - install -D squashfs-root/mqttx.png /app/share/icons/hicolor/512x512/apps/com.emqx.MQTTX.png
+      - install -D com.emqx.MQTTX.desktop /app/share/applications/com.emqx.MQTTX.desktop
```
Comment (hfiguiere) [inline com.emqx.MQTTX.yaml:29]: doesn't the appimage come with its .desktop file?

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "only-arches": ["x86_64"],
```
Comment (hfiguiere) [inline flathub.json:2]: :-(

File: com.emqx.MQTTX.yaml (hunk 1 above)
Comment (ysfscream) [reply com.emqx.MQTTX.yaml:29]: Sorry! I don't seem to know too well. :)

File: flathub.json (hunk 2 above)
Comment (ysfscream) [reply flathub.json:2]: OK. But this may be temporary. 😅

File: com.emqx.MQTTX.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+app-id: com.emqx.MQTTX
+branch: stable
```
Comment (barthalion) [inline com.emqx.MQTTX.yaml:2]:
```suggestion
```

File: com.emqx.MQTTX.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node14
+# Electron doesn't use a traditional locale format
+# so separate-locales is useless.
```
Comment (barthalion) [inline com.emqx.MQTTX.yaml:11]:
```suggestion
```

File: com.emqx.MQTTX.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+separate-locales: false
+command: mqttx.sh
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --filesystem=~/.config/MQTTX
```
Comment (barthalion) [inline com.emqx.MQTTX.yaml:19]: Shouldn't this happen by default? Other Electron apps properly respect `XDG_CONFIG_HOME`.

File: com.emqx.MQTTX.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+  <recommends>
+    <control>pointing</control>
+    <control>keyboard</control>
+  </recommends>
+
+  <description>
+    <p>
+      This is an unofficial package. Issues with it should not be reported to MQTT X upstream!
+    </p>
```
Comment (barthalion) [inline com.emqx.MQTTX.metainfo.xml:50]: Please move this disclaimer to the bottom of `<description>`.

File: com.emqx.MQTTX.yaml (hunk 1 above)
Comment (barthalion) [reply com.emqx.MQTTX.yaml:29]: Looks like it doesn't.

File: com.emqx.MQTTX.yaml (hunk 5 above)
Comment (ysfscream) [reply com.emqx.MQTTX.yaml:19]: So should I remove it from this line?

Comment (ysfscream) [reply com.emqx.MQTTX.yaml:19]: Should be the default.

**Comment by barthalion** [comment]
bot, build com.emqx.MQTTX

**Comment by barthalion** [comment]
```
libGL error: MESA-LOADER: failed to retrieve device information
libGL error: MESA-LOADER: failed to open amdgpu: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/amdgpu_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/GL/default/lib/dri, suffix _dri)
libGL error: failed to load driver: amdgpu
libGL error: failed to open /dev/dri/card0: No such file or directory
libGL error: failed to load driver: radeonsi
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null
```

You should add `--device=dri` to finish-args.

**Comment by barthalion** [comment]
One more question: I've noticed you are the developer working on MQTT X. Is there any reason to make this package unofficial?

**Comment by ysfscream** [comment]
> ```
> libGL error: MESA-LOADER: failed to retrieve device information
> libGL error: MESA-LOADER: failed to open amdgpu: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/amdgpu_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/GL/default/lib/dri, suffix _dri)
> libGL error: failed to load driver: amdgpu
> libGL error: failed to open /dev/dri/card0: No such file or directory
> libGL error: failed to load driver: radeonsi
> libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null
> ```
> 
> You should add `--device=dri` to finish-args.

OK! I will add it.

**Comment by ysfscream** [comment]
> One more question: I've noticed you are the developer working on MQTT X. Is there any reason to make this package unofficial? We usually allow such disclaimers for third-party maintained projects, but it doesn't seem to apply here!

YES, I am working on MQTT X, But this information is from the help of community users, I'm sorry I didn't read it carefully, it's official and I removed the disclaimer.

**Comment by barthalion** [comment]
Thanks!

/merge

**Comment by ysfscream** [comment]
image

Might take some time?

**Comment by barthalion** [comment]
Yeah, it's going to take 3-5 hours before it's actually published to Flathub.

**Comment by ysfscream** [comment]
@barthalion Hi! Sorry to interrupt, I just want to ask where could I see the statistics of downloads?

**Comment by barthalion** [comment]
@ysfscream They are not currently visible on the website, but can be accessed with external tools like https://klausenbusk.github.io/flathub-stats/.

**Comment by ysfscream** [comment]
@barthalion Thanks! This really helps me.

**Pull request closed without merging**